src/compas/CompasImportFromApi.ts

```diff
@@ -0,0 +1,61 @@
+import {customElement, html, LitElement, TemplateResult} from "lit-element";
+
+import '@material/mwc-button';
+
+import {newOpenDocEvent, newWizardEvent} from "../foundation.js";
+
+import {createLogEvent, handleError, handleResponse, parseXml} from "../compas-services/foundation.js";
+import {dispatchEventOnOpenScd} from "./foundation.js";
+
+import '../WizardDivider.js';
+import './CompasSclTypeList.js';
+import './CompasScl.js';
+import { CompasSettings } from "./CompasSettings.js";
+import { CompasCimMappingService } from "../compas-services/CompasCimMappingService.js";
+
+@customElement('compas-import-from-api')
+export default class CompasImportFromApiElement extends LitElement {
+  private async processCimFile(name: string) {
+    const url = CompasSettings().compasSettings.importFromApiUrl;
+
+    const doc = await fetch(url + '/' + name + '.xml')
+    .catch(handleError)
+    .then(handleResponse)
+    .then(parseXml);
+    
+    await CompasCimMappingService().map({cimData: [{name: name + '.xml', doc: doc}]}).then(response => {
+      const sclName = name + ".ssd";
+
+      const sclElement = response.querySelectorAll("SCL").item(0);
```

Comment:
Is it not just possible to retrieve the response.textContent and use that to load it in a Document.

Something like done when opening a file:
    const text = await file.text();
    const docName = file.name;
    const doc = new DOMParser().parseFromString(text, 'application/xml');